### PR TITLE
Scrape config and probe support in target allocator

### PR DESCRIPTION
**Pull request by jaronoff97**
**Description:** Enables support for pulling scrape configs and probes in the target allocator

**Link to tracking Issue(s):**

- Resolves: #1842 

**Testing:** Unit

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by jaronoff97** [comment]
@swiatekm how do we want to handle preventing this being a breaking change https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1beta1/targetallocator_rbac.go#L31-L33, i.e. the TA starts without the rbac it needs and crashes?

**Comment by swiatekm** [comment]
> @swiatekm how do we want to handle preventing this being a breaking change https://github.com/open-telemetry/opentelemetry-operator/blob/main/apis/v1beta1/targetallocator_rbac.go#L31-L33, i.e. the TA starts without the rbac it needs and crashes?

We should treat the fact that it crashes as a bug and ensure it keeps running.  I thought we had an issue for this somewhere.

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/open-telemetry/opentelemetry-operator
 
-go 1.22.0
+go 1.23
+
+toolchain go1.23.2
```
Comment (swiatekm) [inline go.mod:5]: FYI, as per the Go support policy, we can't set 1.23 as the minimum until 1.24 is released.

File: go.mod [hunk 2]
```diff
@@ -2,6 +2,8 @@ module github.com/open-telemetry/opentelemetry-operator
 
 go 1.22.0
 
+toolchain go1.22.5
```
Comment (swiatekm) [inline go.mod:5]: If we're going to suggest a toolchain, how about 1.22.8, which is the latest in the 1.22 line?

File: cmd/otel-allocator/watcher/promOperator_test.go [hunk 3]
```diff
@@ -946,7 +1080,7 @@ func TestNamespaceLabelUpdate(t *testing.T) {
 
 	select {
 	case <-events:
-	case <-time.After(time.Second):
+	case <-time.After(3 * time.Second):
```
Comment (swiatekm) [inline cmd/otel-allocator/watcher/promOperator_test.go:1083]: Can we standardize on 5 seconds for these kinds of tests? They usually complete much faster, and having to wait 5 seconds for a fail is a fair price to pay for lack of random flaky failures on GHA runners.

**Review by swiatekm — APPROVED** [review]
LGTM, minor nitpicks.

**Review by frzifus — APPROVED** [review]

**Pull request merged**